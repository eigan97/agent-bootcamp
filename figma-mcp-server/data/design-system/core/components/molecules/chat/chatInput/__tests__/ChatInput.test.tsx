import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import ChatInput, { ChatInputProps as Props } from '../ChatInput';

const StringValue = 'Sample message';
const FunctionValue = jest.fn();
const BooleanValue = [true, false];

describe('ChatInput component', () => {
  const mapper: Record<string, any> = {
    placeholder: valueHelper('Custom placeholder', { required: true }),
    defaultValue: valueHelper(StringValue, { required: true }),
    disabled: valueHelper(BooleanValue, { required: true, iterate: true }),
    showStopButton: valueHelper(BooleanValue, { required: true, iterate: true }),
    onChange: valueHelper(FunctionValue, { required: true }),
    onClick: valueHelper(FunctionValue, { required: true }),
    onBlur: valueHelper(FunctionValue, { required: true }),
    onFocus: valueHelper(FunctionValue, { required: true }),
    onKeyDown: valueHelper(FunctionValue, { required: true }),
    onSend: valueHelper(FunctionValue, { required: true }),
    onStopGenerating: valueHelper(FunctionValue, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<ChatInput {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('ChatInput component', () => {
  it('renders with default props', () => {
    const { getByPlaceholderText, getByTestId } = render(<ChatInput />);
    expect(getByPlaceholderText('Start typing...')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-SendButton')).toBeInTheDocument();
  });

  it('handles input change correctly', () => {
    const onChange = jest.fn();
    const { getByPlaceholderText } = render(<ChatInput onChange={onChange} />);
    const input = getByPlaceholderText('Start typing...');

    fireEvent.change(input, { target: { value: 'New message' } });
    expect(onChange).toHaveBeenCalled();
  });

  it('handles send button click correctly', () => {
    const onSend = jest.fn();
    const { getByPlaceholderText, getByTestId } = render(<ChatInput onSend={onSend} />);
    const input = getByPlaceholderText('Start typing...');
    const sendButton = getByTestId('DesignSystem-SendButton');

    fireEvent.change(input, { target: { value: 'Test message' } });
    fireEvent.click(sendButton);
    expect(onSend).toHaveBeenCalled();
  });

  it('shows stop button when showStopButton is true', () => {
    const { getByTestId } = render(<ChatInput showStopButton={true} />);
    expect(getByTestId('DesignSystem-StopButton')).toBeInTheDocument();
  });

  it('handles stop generating button click correctly', () => {
    const onStopGenerating = jest.fn();
    const { getByTestId } = render(<ChatInput showStopButton={true} onStopGenerating={onStopGenerating} />);

    const stopButton = getByTestId('DesignSystem-StopButton');
    fireEvent.click(stopButton);
    expect(onStopGenerating).toHaveBeenCalled();
  });

  it('disables input and buttons when disabled prop is true', () => {
    const { getByPlaceholderText, getByTestId } = render(<ChatInput disabled={true} />);
    const input = getByPlaceholderText('Start typing...');
    const sendButton = getByTestId('DesignSystem-SendButton');

    expect(input).toBeDisabled();
    expect(sendButton).toBeDisabled();
  });

  it('renders custom action renderer when provided', () => {
    const actionRenderer = () => <button data-test="DesignSystem-CustomAction">Custom Action</button>;
    const { getByTestId } = render(<ChatInput actionRenderer={actionRenderer} />);

    expect(getByTestId('DesignSystem-CustomAction')).toBeInTheDocument();
  });

  it('clears input after sending message', () => {
    const { getByPlaceholderText, getByTestId } = render(<ChatInput />);
    const input = getByPlaceholderText('Start typing...');
    const sendButton = getByTestId('DesignSystem-SendButton');

    fireEvent.change(input, { target: { value: 'Test message' } });
    fireEvent.click(sendButton);
    expect(input).toHaveValue('');
  });

  describe('className and data-test attributes', () => {
    it('applies custom className to the container', () => {
      const customClass = 'custom-chat-input';
      const { getByTestId } = render(<ChatInput className={customClass} />);
      const chatInputContainer = getByTestId('DesignSystem-ChatInput');
      expect(chatInputContainer).toHaveClass(customClass);
    });

    it('applies correct classes when disabled', () => {
      const { getByTestId } = render(<ChatInput disabled={true} />);
      const chatInputContainer = getByTestId('DesignSystem-ChatInput');
      expect(chatInputContainer).toHaveClass('ChatInput--disabled');
    });

    it('verifies data-test attributes on all interactive elements', () => {
      const { getByTestId } = render(<ChatInput />);

      expect(getByTestId('DesignSystem-ChatInput')).toBeInTheDocument();

      expect(getByTestId('DesignSystem-ChatInput-textarea')).toBeInTheDocument();

      expect(getByTestId('DesignSystem-ChatInput-actions')).toBeInTheDocument();

      expect(getByTestId('DesignSystem-SendButton')).toBeInTheDocument();

      const { getByTestId: getByTestIdWithStop } = render(<ChatInput showStopButton={true} />);
      expect(getByTestIdWithStop('DesignSystem-StopButton')).toBeInTheDocument();
    });

    it('verifies textarea has correct data-test attribute and class', () => {
      const { getByTestId } = render(<ChatInput />);
      const textarea = getByTestId('DesignSystem-ChatInput-textarea');
      expect(textarea).toBeInTheDocument();
      expect(textarea).toHaveClass('ChatInput-textarea');
    });
  });
});
