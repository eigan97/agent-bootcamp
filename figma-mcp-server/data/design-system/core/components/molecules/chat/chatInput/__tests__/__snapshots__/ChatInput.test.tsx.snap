// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput component 
  placeholder: "Custom placeholder", defaultValue: "Sample message", disabled: false, showStopButton: false, onChange: "[Function]", onClick: "[Function]", onBlur: "[Function]", onFocus: "[Function]", onKeyDown: "[Function]", onSend: "[Function]", onStopGenerating: "[Function]"
 1`] = `
<body>
  <div>
    <div
      class="ChatInput"
      data-test="DesignSystem-ChatInput"
    >
      <textarea
        class="ChatInput-textarea"
        data-test="DesignSystem-ChatInput-textarea"
        placeholder="Custom placeholder"
      >
        Sample message
      </textarea>
      <div
        class="ChatInput-actions"
        data-test="DesignSystem-ChatInput-actions"
      >
        <button
          aria-label="Send"
          class="Button Button--tiny Button--tinySquare Button--primary"
          data-test="DesignSystem-SendButton"
          tabindex="0"
        >
          <div
            class="Button-icon"
            data-test="DesignSystem-Button--Icon-Wrapper"
          >
            <i
              class="material-symbols material-symbols-rounded Icon"
              data-test="DesignSystem-Button--Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              arrow_upward_alt
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`ChatInput component 
  placeholder: "Custom placeholder", defaultValue: "Sample message", disabled: false, showStopButton: true, onChange: "[Function]", onClick: "[Function]", onBlur: "[Function]", onFocus: "[Function]", onKeyDown: "[Function]", onSend: "[Function]", onStopGenerating: "[Function]"
 1`] = `
<body>
  <div>
    <div
      class="ChatInput"
      data-test="DesignSystem-ChatInput"
    >
      <textarea
        class="ChatInput-textarea"
        data-test="DesignSystem-ChatInput-textarea"
        placeholder="Custom placeholder"
      >
        Sample message
      </textarea>
      <div
        class="ChatInput-actions"
        data-test="DesignSystem-ChatInput-actions"
      >
        <button
          aria-label="Stop Generating"
          class="Button Button--tiny Button--tinySquare Button--alert"
          data-test="DesignSystem-StopButton"
          tabindex="0"
        >
          <div
            class="Button-icon"
            data-test="DesignSystem-Button--Icon-Wrapper"
          >
            <i
              class="material-symbols material-symbols-rounded Icon"
              data-test="DesignSystem-Button--Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              stop
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`ChatInput component 
  placeholder: "Custom placeholder", defaultValue: "Sample message", disabled: true, showStopButton: false, onChange: "[Function]", onClick: "[Function]", onBlur: "[Function]", onFocus: "[Function]", onKeyDown: "[Function]", onSend: "[Function]", onStopGenerating: "[Function]"
 1`] = `
<body>
  <div>
    <div
      class="ChatInput ChatInput--disabled"
      data-test="DesignSystem-ChatInput"
    >
      <textarea
        class="ChatInput-textarea"
        data-test="DesignSystem-ChatInput-textarea"
        disabled=""
        placeholder="Custom placeholder"
      >
        Sample message
      </textarea>
      <div
        class="ChatInput-actions"
        data-test="DesignSystem-ChatInput-actions"
      >
        <button
          aria-label="Send"
          class="Button Button--tiny Button--tinySquare Button--primary"
          data-test="DesignSystem-SendButton"
          disabled=""
          tabindex="0"
        >
          <div
            class="Button-icon"
            data-test="DesignSystem-Button--Icon-Wrapper"
          >
            <i
              class="material-symbols material-symbols-rounded Icon"
              data-test="DesignSystem-Button--Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              arrow_upward_alt
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`ChatInput component 
  placeholder: "Custom placeholder", defaultValue: "Sample message", disabled: true, showStopButton: true, onChange: "[Function]", onClick: "[Function]", onBlur: "[Function]", onFocus: "[Function]", onKeyDown: "[Function]", onSend: "[Function]", onStopGenerating: "[Function]"
 1`] = `
<body>
  <div>
    <div
      class="ChatInput ChatInput--disabled"
      data-test="DesignSystem-ChatInput"
    >
      <textarea
        class="ChatInput-textarea"
        data-test="DesignSystem-ChatInput-textarea"
        disabled=""
        placeholder="Custom placeholder"
      >
        Sample message
      </textarea>
      <div
        class="ChatInput-actions"
        data-test="DesignSystem-ChatInput-actions"
      >
        <button
          aria-label="Stop Generating"
          class="Button Button--tiny Button--tinySquare Button--alert"
          data-test="DesignSystem-StopButton"
          disabled=""
          tabindex="0"
        >
          <div
            class="Button-icon"
            data-test="DesignSystem-Button--Icon-Wrapper"
          >
            <i
              class="material-symbols material-symbols-rounded Icon"
              data-test="DesignSystem-Button--Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              stop
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</body>
`;
